### PR TITLE
FunctionsAnalyzer - add "isTheSameClassCall" for correct verifying of function calls

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

File: src/Tokenizer/Analyzer/FunctionsAnalyzer.php [hunk 1]
```diff
@@ -96,4 +96,31 @@ public function getFunctionReturnType(Tokens $tokens, $methodIndex)
...
         return new TypeAnalysis($type, $typeStartIndex, $typeEndIndex);
     }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    public function isTheSameClassCall(Tokens $tokens, $index)
```
Comment (SpacePossum) [inline src/Tokenizer/Analyzer/FunctionsAnalyzer.php]: I like the PR as a good clean up, I'm not sure about the name of this method though. 
Also, not sure about adding a new method to this class following the reasoning [here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3961#issuecomment-450569021), maybe a new analyzer would be a better fit?

Comment (kubawerlos) [reply src/Tokenizer/Analyzer/FunctionsAnalyzer.php]: I'm open for suggestion for new names (analyzer and method) ;)

`MethodssAnalyzer`? the function name is the best I can think of.

Comment (julienfalque) [reply src/Tokenizer/Analyzer/FunctionsAnalyzer.php]: This class is `@internal` so I think adding this method here is fine.

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request merged**